src/NHapi.Base/Parser/MessageIterator.cs

```diff
@@ -117,6 +124,18 @@ public virtual bool MoveNext()
 
                 var structureDefinition = currentPosition.StructureDefinition;
 
+                if (parser is PipeParser pipeParser && pipeParser.ParserConfiguration.NonGreedyMode)
+                {
+                    var nonGreedyPosition = CouldBeNonGreedy();
+                    if (nonGreedyPosition != null)
+                    {
+                        while (!Equals(GetCurrentPosition().StructureDefinition, nonGreedyPosition))
```

Comment:
the `==` operator of StructureDefeination has been overloaded so this could be:

```csharp
while (GetCurrentPosition().StructureDefinition != nonGreedyPosition)
```

Which I think reads a little better, thoughts?